lib/components/Charts/BarChart/index.tsx

```diff
@@ -72,6 +86,22 @@ const _BarChart = <K extends ChartConfig>(
           top: label ? 24 : 0,
         }}
         stackOffset={type === "stacked-by-sign" ? "sign" : undefined}
+        onClick={(data) => {
+          if (!onClick || !data.activeLabel || !data.activePayload) {
+            return
+          }
+
+          const chartData = {
+            label: data.activeLabel,
+            values: {},
+          } as ChartDataPoint<K>
+
+          for (const payload of data.activePayload as ActivePayload<K>) {
```

Comment:
is there a way to avoid casting?

Reply:
I'm open to suggestions, I couldn't find a way :(